web/impact/impact/v1/classes/option_analysis.py

```diff
@@ -24,38 +25,81 @@
-CriterionHelper.register_helper(MatchingProgramCriterionHelper,
-                                "matching", "program")
+CriterionHelper.register_helper(
+    JudgeGenderCriterionHelper, "judge", "gender")
+CriterionHelper.register_helper(
+    JudgeRoleCriterionHelper, "judge", "role")
+CriterionHelper.register_helper(
+    MatchingIndustryCriterionHelper, "matching", "industry")
+CriterionHelper.register_helper(
+    MatchingProgramCriterionHelper, "matching", "program")
 
 
 class OptionAnalysis(object):
     _judge_to_count = None
+    expert_profile = "judge__expertprofile__"
+    industry_judge_field = expert_profile + "primary_industry__name"
+    program_judge_field = expert_profile + "home_program_family__name"
+    gender_judge_field = expert_profile + "gender"
+    role_judge_field = expert_profile + "expert_category__name"
 
     def __init__(self,
-                 option_spec,
                  apps,
                  app_ids,
                  judging_round,
                  application_counts,
                  criterion_helpers):
-        self.option_spec = option_spec
+        self.criterion_helpers = criterion_helpers
+        self.option_spec = None
         self.judging_round = judging_round
-        self.helper = CriterionOptionSpecHelper(
-            option_spec,
-            criterion_helpers)
+        self.cycle = self.judging_round.program.cycle
+        self.helper = None
         self.apps = apps
         self.app_ids = app_ids
         self.application_counts = application_counts
+        self.criterion_total_capacities = {}
+        self.judge_to_capacity_cache = None
+        self.criterion_total_functions = {
+            ("judge", "gender"): {
+                'function': self.general_criterion_total_capacity,
+                'filter_field': self.gender_judge_field,
+            },
+            ("reads", "reads"): {
+                'function': None,
+            },
+            ("judge", "role"): {
+                'function': self.general_criterion_total_capacity,
+                'filter_field': self.role_judge_field,
+            },
+            ("matching", "industry"): {
+                'function': self.general_criterion_total_capacity,
+                'filter_field': self.industry_judge_field,
+            },
+            ("matching", "program"): {
+                'function': self.general_criterion_total_capacity,
+                'filter_field': self.program_judge_field,
+            }
+        }
+
+        active_scenarios = Scenario.objects.filter(
+            judging_round=judging_round, is_active=True)
+        self.jpa = JudgeApplicationFeedback.objects.filter(
+            application__in=apps,
+            panel__judgepanelassignment__scenario__in=active_scenarios)
+        self.completed_feedbacks = self.jpa.filter(
+            feedback_status=JUDGING_FEEDBACK_STATUS_COMPLETE)
+        self.application_criteria_read_state_cache = {}
+        self.gender_match_dict = dict(
```

Comment:
This seems to be here because some criteria were set up wrong and have options "female" and "male" instead of "f" and "m". We should just use the db values ("f" and "m") and fix the erroneous criteria options rather than writing the mistake into the code.